### PR TITLE
feat: adds initial support for dictionary APIs

### PR DESCRIPTION
**Pull request by eaddingtonwhite**
Support for following commands added
```
hmset
hget
hset
hgetall
hdel
```

### CODE REVIEW

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

File: test/hash.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  it('should accept a string field and string value pair on hset', async () => {
+    const key = v4();
+    const field = v4();
+    const value = v4();
+    const result = await client.hset(key, field, value);
+    expect(result).toBe(1);
+
+    const getAllResult = await client.hgetall(key);
+    expect(getAllResult).toEqual({[field]: value});
```
Comment (malandis) [inline test/hash.test.ts:21]: Why is the field in an array?

**Review by malandis — COMMENTED** [review]
Looks good just one question about the expect on an object.

**Review by malandis — APPROVED** [review]
:shipit:

**Pull request merged**